File: api/src/main/java/com/openlattice/edm/collection/CollectionTemplateKey.java
package com.openlattice.edm.collection;

import java.util.Objects;
import java.util.UUID;

public class CollectionTemplateKey {

    private UUID entitySetCollectionId;
    private UUID templateTypeId;

    public CollectionTemplateKey( UUID entitySetCollectionId, UUID templateTypeId ) {
        this.entitySetCollectionId = entitySetCollectionId;
        this.templateTypeId = templateTypeId;
    }

    public UUID getEntitySetCollectionId() {
        return entitySetCollectionId;
    }

    public UUID getTemplateTypeId() {
        return templateTypeId;
    }

    @Override public boolean equals( Object o ) {
        if ( this == o )
            return true;
        if ( o == null || getClass() != o.getClass() )
            return false;
        CollectionTemplateKey that = (CollectionTemplateKey) o;
        return Objects.equals( entitySetCollectionId, that.entitySetCollectionId ) &&
                Objects.equals( templateTypeId, that.templateTypeId );
    }

    @Override public int hashCode() {
        return Objects.hash( entitySetCollectionId, templateTypeId );
    }

    @Override public String toString() {
        return "CollectionTemplateKey{" +
                "entitySetCollectionId=" + entitySetCollectionId +
                ", templateTypeId=" + templateTypeId +
                '}';
    }
}

File: api/src/main/kotlin/com/openlattice/collections/CollectionTemplateKey.kt
package com.openlattice.collections

import java.util.UUID

/**
 * A collection template key can be used to point to an entity set id
 * by looking up the collection template type id in the entity set collection's template map
 *
 * @param entitySetCollectionId The id of the entity set collection.
 * @param templateTypeId The id of the collection template type.
 */
data class CollectionTemplateKey(
        val entitySetCollectionId: UUID,
        val templateTypeId: UUID)